### PR TITLE
feat(#505): enable default multi-arch build

### PR DESCRIPTION
**Pull request by nohant**
this is done as a basic configuration for https://github.com/stakater/IngressMonitorController/issues/505
the build is correctly working on my host but i dont know how to test it more :(

the part of the github action i think should be discussed or at least taken care of by someone more expert than me

### CODE REVIEW

**Comment by bnallapeta** [comment]
@nohant 
Instead of having different tags for different platforms for the images, it is easier on the user to just have the same tag and let docker figure out the platform. 
If you take a look at our [Reloader images](https://hub.docker.com/r/stakater/reloader/tags), we have done this already by having the same tag with multi-platform support.
Could you please follow the same standard here too? The changes here look mostly right except the Docker push step. You can refer to [this](https://github.com/stakater/Reloader/blob/master/.github/workflows/push.yaml#L97-L112) one for reference.

**Comment by nohant** [comment]
@bnallapeta you are right im sorry, im too used to how we do that.
Now it should work the same as you said in the other configurations.

**Comment by bnallapeta** [comment]
@nohant The Dockerfile file change you earlier had is still needed as it is currently hardcoded :)
Ref: Reloader [Dockerfile](https://github.com/stakater/Reloader/blob/master/Dockerfile#L29)

P.S.: [Image](https://hub.docker.com/layers/stakater/ingressmonitorcontroller/SNAPSHOT-PR-526-ea1afdc3/images/sha256-e28a75f842c1e9496dfc26e36bd593027237c6c94a5783c17a430e7ea1a512f7?context=explore) is generated for only linux/amd64

**Comment by bnallapeta** [comment]
@nohant Its still not generating multi arch images. I am checking on this. Will get back to you.

**Comment by nohant** [comment]
@bnallapeta don't worry mate, I'm sure i've forgot something :D
thanks for the help, really appreciate.

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -97,28 +96,31 @@ jobs:
...
         cache-to: type=inline
+        platforms: linux/amd64,linux/arm,linux/arm64
         tags: |
-          ${{ env.IMAGE_REPOSITORY }}:${{ steps.generate_tag.outputs.GIT_TAG }}
+          ${{ env.IMAGE_REPOSITORY }}:${{ steps.generate_tag.outputs.new_tag }}
         labels: |
           org.opencontainers.image.source=${{ github.event.repository.clone_url }}
           org.opencontainers.image.created=${{ steps.prep.outputs.created }}
           org.opencontainers.image.revision=${{ github.sha }}
+          org.opencontainers.image.platform=${{ matrix.platform }}
```
Comment (bnallapeta) [inline .github/workflows/pull_request.yml:115]: Don't need anymore since you removed matrix.

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -97,28 +96,31 @@ jobs:
...
 
     - name: Comment on PR
       uses: mshick/add-pr-comment@v2
       if: always()
       env:
        GITHUB_TOKEN: ${{ secrets.STAKATER_GITHUB_TOKEN }}
       with:
-        message-success: '@${{ github.actor }} Image is available for testing. `docker pull ${{ env.IMAGE_REPOSITORY }}:${{ steps.generate_tag.outputs.GIT_TAG }}`'
+        message-success: '@${{ github.actor }} Image is available for testing. `docker pull ${{ env.IMAGE_REPOSITORY }}:${{ steps.generate_tag.outputs.GIT_TAG }}-${{ matrix.platform }}`'
```
Comment (bnallapeta) [inline .github/workflows/pull_request.yml:123]: Same as above.

File: .github/workflows/push.yml [hunk 3]
```diff
@@ -93,8 +93,7 @@ jobs:
     - name: Set up Docker Buildx
       uses: docker/setup-buildx-action@v2
       with:
-        driver-opts: |
-            image=moby/buildkit:v0.9.3
+        driver-opts: 'image=moby/buildkit:v0.11.5'
```
Comment (bnallapeta) [inline .github/workflows/push.yml:96]: Same as above.

File: .github/workflows/push.yml [hunk 4]
```diff
@@ -114,13 +113,16 @@ jobs:
...
+        build-args: BUILD_PARAMETERS=${{ env.BUILD_PARAMETERS }}
         cache-to: type=inline
+        platforms: linux/amd64,linux/arm,linux/arm64
         tags: |
           ${{ env.IMAGE_REPOSITORY }}:${{ steps.generate_tag.outputs.new_tag }}
         labels: |
           org.opencontainers.image.source=${{ github.event.repository.clone_url }}
           org.opencontainers.image.created=${{ steps.prep.outputs.created }}
           org.opencontainers.image.revision=${{ github.sha }}
+          org.opencontainers.image.platform=${{ matrix.platform }}
```
Comment (bnallapeta) [inline .github/workflows/push.yml:125]: Same as above.

**Comment by nohant** [comment]
fixed all the request points, sorry for the matrix leftover.

**Review by bnallapeta — APPROVED** [review]

**Comment by bnallapeta** [comment]
@nohant We now have multi arch support for IMC :) 

https://hub.docker.com/layers/stakater/ingressmonitorcontroller/v2.1.39/images/sha256-f9347ced1826bc88f83ca4091499c706d25188e6b17db80bd697f4fc5d7ff706?context=explore

**Pull request merged**